ext/dom/node.c

```diff
@@ -876,6 +878,7 @@ PHP_METHOD(DOMNode, insertBefore)
 	}
 
 	if (child->type == XML_DOCUMENT_FRAG_NODE && child->children == NULL) {
+		// Todo convert to Error?
```

Comment:
I would say no, i feel its even wrong to throw a warning here. inserting something empty before a node should just do nothing.